docs/Privacy/Privacy.md

```diff
@@ -0,0 +1,107 @@
+ what information related to them may be collected and stored and by
+ whom and to whom that information may be disclosed.” For the purposes
+ of this Specification, this right of privacy also applies to
+ organizations to the extent permitted by law.
+- **Private State**:   A data that is not shared in the clear in the
+ globally replicated state tree. This data can represent bilateral or
+ multilateral arrangements between counterparties, for example in
+ private transactions.
+- **Private Transaction**: A transaction where some information about
+ the transaction, such as the payload data, or the sender and
+ recipient, is only available to the subset of network actors who are
+ parties to that transaction.
+- **Private Transaction Manager**: A subsystem of an Enterprise Ethereum
+  system for implementing privacy and permissioning known as Enclave.
+  e.g. [Orion](https://github.com/PegaSysEng/orion) .
+- **Actor**:   The public key of someone involved in the private transaction (as per the privateFor member).
+- **Privacy group**:   The actors that are involved in a set of transactions.
+
+
+### How Private Transaction is Processed
+
+1. DApp sends a private transaction through [eea_sendrawtransaction](../Reference/JSON-RPC-API-Methods.md#eea_sendRawTransaction),
+  uses the `privateFor` attribute to specify the list of actors,
+  `privateFrom` with the sender public key, and `restriction` to handle
+  with *restricted* and *unrestricted* transaction.
+ 
+2. The JSON RPC detects the private transaction and hands it off to a
+   Private Transaction Handler.
+
+3. The Private Transaction Handler connect to the Enclave (Private
+   Transaction Manager) instance and sends the transaction through this
+   instance.
+
+4. Enclave will authenticated-encrypt and distribute the private
+   transaction through the nodes, including the actors, and an id for
+   the privacy group that it has generated. Each Enclave node will store
+   the transaction, along with the associated privacy group and actor
+   list.
+
+5. Enclave sends the Private Transaction Hash back to the Private
+  Transaction Handler (as a response to the synchronous HTTP call),
+  along with the privacy group.
+ 
+6. The Private Transaction Handler will lookup the address of the
+  Privacy Precompile Smart Contract and will create and send a
+  transaction to this Precompiled contract, including the Transaction
+  Hash from the Enclave as a parameter. This transaction will be known
+  as a Privacy Marker Transaction.
+  -  The Privacy Marker Transaction will be signed using a Private
+      Transaction Signing Key.
+
+7. The Transaction Processor will process the Privacy Marker Transaction
+  as per any other transaction. When this transaction executes, the
+  Privacy Precompile code will be passed the Private Transaction Hash
+  from Enclave.
+
+8. The Privacy Precompile will ask the Enclave for the Private
+  Transaction and Privacy Group based on the Private Transaction Hash.
+
```

Comment:
Is this when a public transaction containing a Privacy Marker Transaction is received?  It's not clear what events causes the Privacy Precompile to request the Private Transaction and Privacy Group from the Enclave at this point. If this is the point where the public transaction containing a Privacy Marker is received, it may be clearer to split this out into a separate process.  ie, have sections on both 'How a Private Transaction is Created and Sent' and 'How a Transaction containing a Privacy Marker Transaction is Processed'

Reply:
sorted